src/gateway/environments/jsdom-sixteen/jsdom-sixteen-resource-loader.js

```diff
@@ -174,7 +174,30 @@ export class JSDOMSixteenResourceLoader extends ResourceLoader {
                  */
                 return Buffer.from("");
             }
-            return Buffer.from(response.text);
+
+            /**
+             * If buffering is unset (which is better for memory use), then
+             * the text field is only present for things that are parsed
+             * (like JSON). Otherwise, we want response body.
+             *
+             * If buffering is set to false (which is even better for memory),
+             * then the text field is never present and we have to use body.
+             *
+             * If buffering is on (bad for memory use), then text will always
+             * be present.
+             *
+             * So, let's cater to almost all eventualities.
+             *
+             * There are some circumstances where body might be null too, but
+             * those are cases we should never encounter.
+             */
+            const data = response.text == null ? response.body : response.text;
```

Comment:
Note that we could use `response.buffered` to determine the buffered state, but this doesn't tell us if any of this is valid. We could also console `response.xhr.response` directly.

Reply:
This is not relevant any more. Resolving.